core/src/tests/bsock_mock.h

```diff
@@ -103,7 +103,7 @@ BareosSocketMock::~BareosSocketMock()
     errmsg = nullptr;
   }
   if (who_) { /* duplicated */
-    free(who_);
+    free(const_cast<char*>(who_));
```

Comment:
Bad bad practice, see previous comment.

Reply:
I rolled back the complete commit where who_ was declared from char* to const char*.